mewarpx/mewarpx/diags_store/particle_histogram_diagnostic.py

```diff
@@ -12,127 +12,91 @@
 
 logger = logging.getLogger(__name__)
 
-class ParticleHistDiag(diag_base.WarpXDiagnostic):
 
-    """Record a per-shape multi-dimensional histogram of particle properties.
+class BaseParticleHistDiag(diag_base.WarpXDiagnostic):
+
+    """Base class for diagnostics that record a multi-dimensional histogram of
+    particle properties.
```

Comment:
Add into here or __init__ documentation the things the subclass needs to define - `linres` and `domain` at least I believe?